src/Tools/dotnet-user-secrets/src/Internal/ProjectIdResolver.cs

```diff
@@ -53,26 +53,51 @@ public string Resolve(string project, string configuration)
                     "/p:Configuration=" + configuration,
                     "/p:CustomAfterMicrosoftCommonTargets=" + _targetsFile,
                     "/p:CustomAfterMicrosoftCommonCrossTargetingTargets=" + _targetsFile,
+                    "-verbosity:detailed",
```

Comment:
Isn't this like a lot bigger? What is the size of the output when enabling this?

Reply:
None, because it only shows up for errors

Reply:
Yes, but when an error occurs, what is the size of the file or output that is written? Verbose logs for msbuild are very verbose IIRC.

Reply:
This is detailed, not diagnostic. I can give you an example in a few minutes.

Reply:
```
Build started 3/2/2020 5:41:58 PM.
     0>Resolving SDK 'Microsoft.NET.Sdk'...
       Property reassignment: $(MSBuildProjectExtensionsPath)="C:\Users\Brennan\AppData\Local\Temp\usersecretstest\2a370e6c-51be-414b-a236-a093b60bcdd8\obj\" (previous value: "obj\") at E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Current\Microsoft.Common.props (56,5)
       The "Configuration" property is a global property, and cannot be modified.
       The "Configuration" property is a global property, and cannot be modified.
       Property reassignment: $(TargetsForTfmSpecificContentInPackage)=";PackTool;_PackProjectToolValidation" (previous value: ";PackTool") at E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.PackProjectTool.props (15,5)
       Property reassignment: $(OutputType)="Exe" (previous value: "Library") at C:\Users\Brennan\AppData\Local\Temp\usersecretstest\2a370e6c-51be-414b-a236-a093b60bcdd8\TestProject.csproj (3,5)
       The "Configuration" property is a global property, and cannot be modified.
       Property reassignment: $(MSBuildAllProjects)=";E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\NuGet.targets;E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Current\Microsoft.Common.CrossTargeting.targets\ImportAfter\Microsoft.TestPlatform.CrossTargeting.targets" (previous value: ";E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\NuGet.targets") at E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Current\Microsoft.Common.CrossTargeting.targets\ImportAfter\Microsoft.TestPlatform.CrossTargeting.targets (15,5)
       Property reassignment: $(MSBuildAllProjects)=";E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\NuGet.targets;E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Current\Microsoft.Common.CrossTargeting.targets\ImportAfter\Microsoft.TestPlatform.CrossTargeting.targets;E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets" (previous value: ";E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\NuGet.targets;E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Current\Microsoft.Common.CrossTargeting.targets\ImportAfter\Microsoft.TestPlatform.CrossTargeting.targets") at E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets (18,5)
       Property reassignment: $(PackageDescription)="Package Description" (previous value: "") at E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets (32,5)
       Property reassignment: $(GenerateNuspecDependsOn)="Build;_LoadPackInputItems; _GetTargetFrameworksOutput; _WalkEachTargetPerFramework; _GetPackageFiles; " (previous value: "_LoadPackInputItems; _GetTargetFrameworksOutput; _WalkEachTargetPerFramework; _GetPackageFiles; ") at E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets (55,5)
       Property reassignment: $(MSBuildAllProjects)="C:\Users\Brennan\AppData\Local\Temp\usersecretstest\2a370e6c-51be-414b-a236-a093b60bcdd8\TestProject.csproj;;E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\NuGet.targets;E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Current\Microsoft.Common.CrossTargeting.targets\ImportAfter\Microsoft.TestPlatform.CrossTargeting.targets;E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets" (previous value: ";E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\NuGet.targets;E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Current\Microsoft.Common.CrossTargeting.targets\ImportAfter\Microsoft.TestPlatform.CrossTargeting.targets;E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets") at 
       The target "_CheckForInvalidConfigurationAndPlatform" listed in a BeforeTargets attribute at "E:\ASPNETCORE\aspnetcore\.dotnet\sdk\5.0.100-preview.2.20120.3\NuGet.targets (463,11)" does not exist in the project, and will be ignored.
     1>Project "C:\Users\Brennan\AppData\Local\Temp\usersecretstest\2a370e6c-51be-414b-a236-a093b60bcdd8\TestProject.csproj" on node 1 (_ExtractUserSecretsMetadata target(s)).
     1>Building with tools version "Current".
       Project file contains ToolsVersion="15.0". This toolset may be unknown or missing, in which case you may be able to resolve this by installing the appropriate version of MSBuild, or the build may have been forced to a particular ToolsVersion for policy reasons. Treating the project as if it had ToolsVersion="Current". For more information, please see http://go.microsoft.com/fwlink/?LinkId=293424.
     1>Target "_ExtractUserSecretsMetadata" in file "E:\ASPNETCORE\aspnetcore\artifacts\bin\dotnet-user-secrets.Tests\Debug\netcoreapp5.0\assets\SecretManager.targets" from project "C:\Users\Brennan\AppData\Local\Temp\usersecretstest\2a370e6c-51be-414b-a236-a093b60bcdd8\TestProject.csproj" (entry point):
       Using "WriteLinesToFile" task from assembly "Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
       Task "WriteLinesToFile"
       Done executing task "WriteLinesToFile".
     1>Done building target "_ExtractUserSecretsMetadata" in project "TestProject.csproj".
     1>Done Building Project "C:\Users\Brennan\AppData\Local\Temp\usersecretstest\2a370e6c-51be-414b-a236-a093b60bcdd8\TestProject.csproj" (_ExtractUserSecretsMetadata target(s)).
Build succeeded.
    0 Warning(s)
    0 Error(s)
Time Elapsed 00:00:00.79
```

Reply:
Detailed should be OK. Agreed that Diagnostic would be tooooooooo much